### ISSUE TITLE
test failures on Alpine Linux edge since 0.12.0 release

### ISSUE THREAD

**Issue by ncopa** [flexmock/flexmock#157]
30 tests fails with the 0.12.0 release:

```
=================================================== test session starts ===================================================
platform linux -- Python 3.11.8, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/ncopa/aports/community/py3-flexmock/src/flexmock-0.12.0
configfile: pyproject.toml
plugins: cov-4.1.0, xdist-3.5.0, mock-3.10.0, requests-mock-1.11.0
collected 236 items                                                                                                       

tests/test_pytest.py ...F.........F.....................................FF.FF.FFF.....F............................ [ 39%]
..........................................................................................FF..FF..FFFFF.FF....FFFF. [ 88%]
....F.......F......F.F.xFxx                                                                                         [100%]

======================================================== FAILURES =========================================================
_____________________________________ test_flexmock_teardown_works_with_pytest_part2 ______________________________________

    def test_flexmock_teardown_works_with_pytest_part2():
        mock = flexmock().should_receive("method2").ordered().mock()
        # Raises CallOrderError if flexmock teardown is not automatically called
        # after test part 1 above
>       mock.method2()

tests/test_pytest.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
build/lib/flexmock/_api.py:508: in mock_method
    expectation = FlexmockContainer.get_flexmock_expectation(self, name, arguments)
build/lib/flexmock/_api.py:1324: in get_flexmock_expectation
    cls._verify_call_order(found, args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'flexmock._api.FlexmockContainer'>, expectation = <flexmock._api.Expectation object at 0x7f0d170506d0>
args = {'kargs': (), 'kwargs': {}}

    @classmethod
    def _verify_call_order(cls, expectation: Expectation, args: Dict[str, Any]) -> None:
        next_method = cls.ordered.pop(0)
        cls.last = next_method
        if expectation is not next_method:
>           raise CallOrderError(
                f"{_format_args(str(expectation._name), args)} called before "
                f"{_format_args(str(next_method._name), next_method._args)}"
            )
E           flexmock.exceptions.CallOrderError: method2() called before method1()

build/lib/flexmock/_api.py:1332: CallOrderError

...
.... <cutted> ...
...

tests/test_pytest.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib/python3.11/contextlib.py:158: in __exit__
    self.gen.throw(typ, value, traceback)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

expected_exception = <class 'flexmock.exceptions.MethodCallError'>
match = 'foo() expected to be called exactly 1 time, called 0 times'

    @contextmanager
    def assert_raises(expected_exception: Type[BaseException], match: Union[RE_TYPE, str, None]):
        """Assert that code raises the correct exception with a correct error message.
    
        Args:
            expected_exception: Type of the expected exception.
            match: String or pattern to match the error message against. Use None
              to skip error message checking.
        """
        try:
            yield
        except Exception as raised_exception:
            if not isinstance(raised_exception, expected_exception):
                raise AssertionError(
                    f"Expected exception '{type(expected_exception)}' "
                    f"but '{type(raised_exception)}' was raised"
                ) from raised_exception
            if match is not None:
                fail = False
                if isinstance(match, RE_TYPE):
                    fail = not match.search(str(raised_exception))
                    match = match.pattern
                else:
                    fail = str(raised_exception) != str(match)
                if fail:
>                   raise AssertionError(
                        f"Expected error message:\n{'-'*39}\n'{str(match)}'\n"
                        f"\nBut got:\n\n'{str(raised_exception)}'\n{'-'*39}\n"
                    ) from raised_exception
E                   AssertionError: Expected error message:
E                   ---------------------------------------
E                   'foo() expected to be called exactly 1 time, called 0 times'
E                   
E                   But got:
E                   
E                   'class_method_with_args("red") expected to be called exactly 1 time, called 0 times'
E                   ---------------------------------------

tests/utils.py:34: AssertionError
_______________________________ TestFailureOnException.test_flexmock_raises_if_no_exception _______________________________
[XPASS(strict)] 
================================================= short test summary info =================================================
FAILED tests/test_pytest.py::test_flexmock_teardown_works_with_pytest_part2 - flexmock.exceptions.CallOrderError: method2() called before method1()
FAILED tests/test_pytest.py::TestForPytest::test_flexmock_teardown_called_between_tests_part2 - flexmock.exceptions.CallOrderError: method2() called before method1()
FAILED tests/test_pytest.py::TestForPytest::test_spy_proxied_derived_class - flexmock.exceptions.MethodCallError: class_method() expected to be called exactly 2 times, called 3 times
FAILED tests/test_pytest.py::TestForPytest::test_spy_proxied_derived_class_with_args - flexmock.exceptions.MethodCallError: class_method_with_args(a="a") expected to be called exactly 2 times, called 3 times
FAILED tests/test_pytest.py::TestForPytest::test_spy_proxied_class - flexmock.exceptions.MethodCallError: class_method() expected to be called exactly 2 times, called 3 times
FAILED tests/test_pytest.py::TestForPytest::test_spy_proxied_class_with_args - flexmock.exceptions.MethodCallError: class_method_with_args("a") expected to be called exactly 2 times, called 3 times
FAILED tests/test_pytest.py::TestForPytest::test_flexmock_ordered_worked_after_default_stub - AssertionError: Expected error message:
FAILED tests/test_pytest.py::TestForPytest::test_flexmock_ordered_works_with_same_args - flexmock.exceptions.CallOrderError: bar() called before bar("a")
FAILED tests/test_pytest.py::TestForPytest::test_flexmock_ordered_works_with_same_args_after_default_stub - flexmock.exceptions.CallOrderError: bar() called before bar("b")
FAILED tests/test_pytest.py::TestForPytest::test_calling_deprecated_properties_should_raise_exception - AssertionError: Expected error message:
FAILED tests/test_pytest.py::TestForPytest::test_mock_class_method_on_derived_class - TypeError: Mock._create_mock_method.<locals>.mock_method() missing 1 required positional argument: 'runtime_self'
FAILED tests/test_pytest.py::TestForPytest::test_mock_class_method_on_derived_class_after_mocking_base_class - TypeError: Mock._create_mock_method.<locals>.mock_method() missing 1 required positional argument: 'runtime_self'
FAILED tests/test_pytest.py::TestForPytest::test_mock_class_method_with_args_on_derived_class - flexmock.exceptions.MethodSignatureError: Arguments for call class_method_with_args did not match expectations:
FAILED tests/test_pytest.py::TestForPytest::test_mock_class_method_with_args_on_derived_class_after_mocking_base_class - flexmock.exceptions.MethodSignatureError: Arguments for call class_method_with_args did not match expectations:
FAILED tests/test_pytest.py::TestForPytest::test_spy_class_method_on_derived_class - KeyError: <flexmock._api.Mock object at 0x7f0d1730ad10>
FAILED tests/test_pytest.py::TestForPytest::test_spy_class_method_on_derived_class_after_spying_base_class - TypeError: Mock._create_mock_method.<locals>.mock_method() missing 1 required positional argument: 'runtime_self'
FAILED tests/test_pytest.py::TestForPytest::test_spy_static_method_on_derived_class - TypeError: 'NoneType' object is not callable
FAILED tests/test_pytest.py::TestForPytest::test_spy_static_method_on_derived_class_after_spying_base_class - TypeError: 'NoneType' object is not callable
FAILED tests/test_pytest.py::TestForPytest::test_spy_class_method_with_args_on_derived_class - KeyError: <flexmock._api.Mock object at 0x7f0d17309ed0>
FAILED tests/test_pytest.py::TestForPytest::test_spy_static_method_with_args_on_derived_class - TypeError: 'NoneType' object is not callable
FAILED tests/test_pytest.py::TestForPytest::test_spy_static_method_with_args_on_derived_class_after_spying_base_class - AssertionError: Expected exception '<class 'type'>' but '<class 'TypeError'>' was raised
FAILED tests/test_pytest.py::TestForPytest::test_is_class_method - AssertionError: assert False is True
FAILED tests/test_pytest.py::TestForPytest::test_is_class_method_proxied - assert False is True
FAILED tests/test_pytest.py::TestForPytest::test_is_static_method - AssertionError: assert False is True
FAILED tests/test_pytest.py::TestForPytest::test_is_static_method_proxied - assert False is True
FAILED tests/test_pytest.py::TestForPytest::test_support_at_least_and_at_most_together - AssertionError: Expected error message:
FAILED tests/test_pytest.py::TestForPytest::test_module_level_function_with_kwargs - AssertionError: Expected error message:
FAILED tests/test_pytest.py::TestForPytest::test_with_args_with_class_method - flexmock.exceptions.MethodSignatureError: Arguments for call class_method_with_args did not match expectations:
FAILED tests/test_pytest.py::TestForPytest::test_class_level_test_for_pytest - AssertionError: Expected error message:
FAILED tests/test_pytest.py::TestFailureOnException::test_flexmock_raises_if_no_exception
======================================== 30 failed, 203 passed, 3 xfailed in 0.68s ========================================
```

The 0.11.3 release has only a single failing test:
```
=================================================== test session starts ===================================================
platform linux -- Python 3.11.8, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/ncopa/aports/community/py3-flexmock/src/flexmock-0.11.3
configfile: pyproject.toml
plugins: cov-4.1.0, xdist-3.5.0, mock-3.10.0, requests-mock-1.11.0
collected 0 items / 1 error                                                                                               

========================================================= ERRORS ==========================================================
__________________________________________ ERROR collecting tests/test_pytest.py __________________________________________
tests/test_pytest.py:7: in <module>
    from flexmock import flexmock
build/lib/flexmock/__init__.py:2: in <module>
    from flexmock import _integrations  # pylint: disable=unused-import
build/lib/flexmock/_integrations.py:101: in <module>
    saved_pytest = runner.call_runtest_hook
E   AttributeError: module '_pytest.runner' has no attribute 'call_runtest_hook'
================================================= short test summary info =================================================
ERROR tests/test_pytest.py - AttributeError: module '_pytest.runner' has no attribute 'call_runtest_hook'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================== 1 error in 0.05s =====================================================
```

**Comment by ollipa** [comment]
It seems that the test collection fails with 0.11.3 because it doesn't support Pytest 8.1. Pytest 8.1 support was fixed in flexmock 0.12.0. What kind of results you get when you run the tests with Pytest version 8.0.2?

**Comment by Cellesti** [comment]
I think this error is seen when building with setuptools instead of poetry-core.

There is a [poetry setting](https://github.com/flexmock/flexmock/blob/0.12.0/pyproject.toml#L52) in `pyproject.toml` which makes pytest use the flexmock plugin, and this is not active when building with setuptools.

I managed to get tests working when building with setuptools by adding `addopts = ["-p flexmock._pytest_plugin"]` under [the options for pytest](https://github.com/flexmock/flexmock/blob/0.12.0/pyproject.toml#L128).

**Comment by ollipa** [comment]
Thanks @Cellesti for pointing out the problem. `setup.py` was missing entrypoint configuration for Flexmock Pytest plugin. Fixed in #158.
